### ISSUE TITLE
Update screen-reader documention (support article, user guide)

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#12363]
### System details

    RStudio Edition : User Guide / Support articles
    RStudio Version : Elsbeth Geranium
    OS Version      : any
    R Version       : any

### Steps to reproduce the problem

In this support article: https://support.posit.co/hc/en-us/articles/360045612413-RStudio-Screen-Reader-Support, and repeated in the user guide (https://docs.posit.co/ide/user/ide/guide/accessibility/screen-reader.html):

> RStudio Desktop
> Notes: Due to underlying accessibility issues introduced by components used to build RStudio Desktop, we cannot yet recommend its use via a screen reader except in an experimental fashion. We are actively working with the developer of these components to get these issues addressed, and will be releasing updates with these fixes as soon as we can.

### Describe the problem in detail

This is no longer true for Electron-based desktop IDE which now has equivalent screen reader support to RStudio Server (still plenty of issues, but at least now future screen-reader fixes will apply equally to desktop and server).

### Describe the behavior you expected

Documention should mention that open-source Desktop now support screen reader, but RDP won't until a future release.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
One difference, which I didn't catch earlier: In RStudio Server, the shortcut is still `Alt-Ctrl-/`, so we need to distinguish between RStudio Desktop and Server for MacOS.

**Comment by gtritchie** [comment]
The shortcut has also changed for Server if you are viewing it from a Mac. Shortcuts depend on what OS the user is on (web browser).

**Comment by ronblum** [comment]
Verified changes for the shortcut on MacOS to both the RStudio User Guide and the support article.
